### PR TITLE
Textarea v2 adding a minimum height

### PR DESCRIPTION
**Pull request by jwoudenberg**
#### What is this

This is a new version of the `TextArea` widget. It adds the option to set a minimum height for text areas and does some initial work in porting the widget to `Html.Styled` (API is changed, internals still rely on some old fashion styling).

### CODE REVIEW

File: src/Nri/Ui/InputStyles.elm [hunk 1]
```diff
@@ -246,3 +251,38 @@ type alias Assets r =
         , icons_searchGray_svg : Asset
         , icons_xBlue_svg : Asset
     }
+
+
+inputPaddingVertical : Px
+inputPaddingVertical =
+    px 8
```
Comment (jwoudenberg) [inline src/Nri/Ui/InputStyles.elm]: 😻

File: src/Nri/Ui/TextArea/V2.elm [hunk 2]
```diff
@@ -50,26 +51,31 @@ type alias Model compatible msg =
...
     , label : String
     , showLabel : Bool
-    , minimumHeight : LengthOrMinMaxDimension compatible
+    , minimumHeight : MinimumHeight
     }
 
 
+type MinimumHeight
+    = DefaultHeight
+    | SingleLine
```
Comment (jwoudenberg) [inline src/Nri/Ui/TextArea/V2.elm]: 💯

File: styleguide-app/assets/TextArea.js [hunk 3]
```diff
@@ -43,7 +43,7 @@ Object.defineProperties(AutoresizeTextArea.prototype, {
   }
 });
 
-AutoresizeTextArea.prototype._onInput = function() {
+AutoresizeTextArea.prototype._resize = function() {
```
Comment (jwoudenberg) [inline styleguide-app/assets/TextArea.js]: Thanks for pulling in these changes from the `TextArea.js` files from the monolith.

File: src/Nri/Ui/TextArea/V2.elm [hunk 4]
```diff
@@ -45,15 +44,23 @@ type alias Model msg =
...
-    , minimumHeight : MinimumHeight
     }
 
 
-type MinimumHeight
+{-| Control whether to auto-expand the height.
+-}
+type HeightBehavior
+    = Fixed
+    | AutoResize Height
```
Comment (jwoudenberg) [inline src/Nri/Ui/TextArea/V2.elm]: 😻 Even better!

File: src/Nri/Ui/TextArea/V2.elm [hunk 5]
```diff
@@ -211,15 +211,7 @@ calculateMinHeight textAreaStyle specifiedHeight =
                     singleLineHeight
 
         DefaultHeight ->
-            case textAreaStyle of
-                DefaultStyle ->
-                    InputStyles.textAreaHeight
-
-                WritingStyle ->
-                    InputStyles.writingMinHeight
```
Comment (ento) [inline src/Nri/Ui/TextArea/V2.elm]: A Writing textarea is still specified to have a min-height of 150px: https://github.com/NoRedInk/noredink-ui/pull/65/files#diff-603eb374d349e2a301c6d44684f6016fR144

Should we use that 150px here as well, or was the distinction of writing vs other styles no longer needed when it's set to auto-resize?

Comment (jwoudenberg) [reply src/Nri/Ui/TextArea/V2.elm]: Good catch, thanks! Will push a fix.

Comment (jwoudenberg) [reply src/Nri/Ui/TextArea/V2.elm]: Added back the height of 150px for text areas. Also checked the styleguide app to see the min-height is applied correctly. Seems to look good!

**Comment by ento** [comment]
published as 5.3.0

**Pull request merged**